nullaway/src/main/java/com/uber/nullaway/handlers/OptionalEmptinessHandler.java

```diff
@@ -168,35 +231,54 @@ public boolean includeApInfoInSavedContext(AccessPath accessPath, VisitorState s
-      Context context,
-      MethodInvocationNode node,
-      Types types,
-      AccessPath.AccessPathContext apContext,
-      AccessPathNullnessPropagation.Updates bothUpdates) {
+  private Optional<MethodInvocationNode> getMethodWrappedByAssertThat(MethodInvocationNode node) {
     Node receiver = node.getTarget().getReceiver();
     if (receiver instanceof MethodInvocationNode) {
       MethodInvocationNode receiverMethod = (MethodInvocationNode) receiver;
       Symbol.MethodSymbol receiverSymbol = ASTHelpers.getSymbol(receiverMethod.getTree());
-      if (methodNameUtil.isMethodAssertThat(receiverSymbol)) {
-        // assertThat will always have at least one argument, So safe to extract from the arguments
+      if (receiverMethod.getArguments().size() == 1
+          && methodNameUtil.isMethodAssertThat(receiverSymbol)) {
         Node arg = receiverMethod.getArgument(0);
         if (arg instanceof MethodInvocationNode) {
-          // Since assertThat(a.isPresent()) changes to
-          // Truth.assertThat(Boolean.valueOf(a.isPresent()))
-          // need to be unwrapped from Boolean.valueOf
-          Node unwrappedArg = ((MethodInvocationNode) arg).getArgument(0);
-          if (unwrappedArg instanceof MethodInvocationNode) {
-            MethodInvocationNode argMethod = (MethodInvocationNode) unwrappedArg;
-            Symbol.MethodSymbol argSymbol = ASTHelpers.getSymbol(argMethod.getTree());
-            if (optionalIsPresentCall(argSymbol, types)) {
-              updateNonNullAPsForOptionalContent(
-                  context, bothUpdates, argMethod.getTarget().getReceiver(), apContext);
-            }
-          }
+          MethodInvocationNode wrappedMethodInvocation = ((MethodInvocationNode) arg);
+          return Optional.of(maybeUnwrapBooleanValueOf(wrappedMethodInvocation));
+        }
+      }
+    }
+    return Optional.empty();
+  }
+
+  private Optional<Node> getVariableOrFieldWrappedByAssertThat(MethodInvocationNode node) {
+    Node receiver = node.getTarget().getReceiver();
+    if (receiver instanceof MethodInvocationNode) {
+      MethodInvocationNode receiverMethod = (MethodInvocationNode) receiver;
+      Symbol.MethodSymbol receiverSymbol = ASTHelpers.getSymbol(receiverMethod.getTree());
+      if (receiverMethod.getArguments().size() == 1
+          && methodNameUtil.isMethodAssertThat(receiverSymbol)) {
+        // assertThat will always have at least one argument, So safe to extract from the arguments
+        Node arg = receiverMethod.getArgument(0);
+        if (arg instanceof LocalVariableNode || arg instanceof FieldAccessNode) {
+          return Optional.of(arg);
+        }
+      }
+    }
+    return Optional.empty();
+  }
+
+  private MethodInvocationNode maybeUnwrapBooleanValueOf(MethodInvocationNode node) {
+    // Since assertThat(a.isPresent()) changes to
+    // Truth.assertThat(Boolean.valueOf(a.isPresent()))
+    // need to be unwrapped from Boolean.valueOf
```

Comment:
this looks odd but was already present in the old code and Truth-tests fail without it

Reply:
ah i see thanks, due to the comment i thought this is some Truth specific thing (idk that library), but yeah i it's just auto boxing by the java compiler :+1: